bin/hdfcoinc/pycbc_coinc_statmap

```diff
@@ -133,20 +212,301 @@ logging.info("calculating ifar/fap values")
 if fore_locs.sum() > 0:
     ifar = background_time / (fnlouder + 1)
     fap = 1 - numpy.exp(- coinc_time / ifar)
-    f['foreground/ifar'] = sec_to_year(ifar)
-    f['foreground/fap'] = fap
+    f['foreground_h/h_iter' + h_iter_string + '/ifar'] = sec_to_year(ifar)
+    f['foreground_h/h_iter' + h_iter_string + '/fap'] = fap
 
     ifar_exc = background_time_exc / (fnlouder_exc + 1)
     fap_exc = 1 - numpy.exp(- coinc_time_exc / ifar_exc)
     f['foreground/ifar_exc'] = sec_to_year(ifar_exc)
     f['foreground/fap_exc'] = fap_exc
 else:
-    f['foreground/ifar'] = numpy.array([])
-    f['foreground/fap'] = numpy.array([])
-    f['foreground/ifar_exc'] = numpy.array([])
-    f['foreground/fap_exc'] = numpy.array([])
+    f['foreground_h/h_iter' + h_iter_string + '/ifar'] = numpy.array([])
+    f['foreground_h/h_iter' + h_iter_string + '/fap'] = numpy.array([])
+    f['foreground_h/h_iter' + h_iter_string + '/ifar_exc'] = numpy.array([])
+    f['foreground_h/h_iter' + h_iter_string + '/fap_exc'] = numpy.array([])
+
+if 'name' in all_trigs.attrs:
+    f.attrs['name'] = all_trigs.attrs['name']
+
+# Incorporate hierarchical removal for any other loud triggers
+logging.info("Beginning hierarchical removal of foreground triggers.")
+
+# Step 1: Check to see if there are triggers that are louder
+#         then all of the inclusive background.
+
+# fnlouder is the number of background events that are louder than
+# the foreground trigger. So if a trigger is louder than all of
+# the background, fnlouder = 0.
+
+loud_t_ind = []
+for i in range(0, len(fnlouder)):
+    if fnlouder[i] == 0:
+        loud_t_ind.append(i)
+
+# Create arrays to save far and fap of hierarchically removed
+# foreground triggers
+
+save_ifar_fore_trig = numpy.array([], dtype=float)
```

Comment:
I can't see the full context from the code block, but note that we shouldn't overwrite the content of files as the they can't be tracked in the workflow. If the contents of an input file changes, then the data should be written to an output file with a new name.